### PR TITLE
[#1365] Raise sync batch size outside of sandblasting to 1000

### LINKED ISSUE #1365: Raise the batch size outside of sandblasting to 1000
Currently `SYNC_BATCH_SIZE` is set to 100. This is fine for scanning during the sandblasting region of the chain, but outside that we should raise the limit to 1000.

Eventually the batch size will be dynamic in number of blocks (instead based on subtree boundaries, which are related to the number of outputs which is where our size and scanning costs come from), but this is a quick intermediate step.

Sandblasting period (based on the dates corresponding to block heights; sandblasting started in late June 2022 and mostly ended in mid-April 2023): 1,710,000 to 2,050,000

### PR DESCRIPTION
**Pull request by HonzaR**
Closes #1365

Sync speed results
---
**New wallet (cca 22K blocks):**

_Before_:
1. 14:15:53.199 - 14:09:02.966 = 6 minutes, 50 seconds, and 233
2. 14:24:15.356 - 14:18:32.068 = 5 minutes, 43 seconds, and 288

_After_:
1. 14:01:34.786 - 13:55:54.256 = 5 minutes, 40 seconds, and 530
2. 15:18:51.461 - 15:14:26.988 = 4 minut, 24 sekund a 473

**Wallet restoration (cca 117K blocks):**

_Before:_
16:32:09.854 - 16:02:00.592 = 30 minut, 9 sekund a 262

_After:_
15:51:07.254 - 15:22:28.317 = 28 minut, 38 sekund a 937

> **Note**
>This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] **Self-review** your own code in GitHub's web interface[^1]
- [ ] Add **automated tests** as appropriate
- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^2] as appropriate
- [ ] Check the **code coverage**[^3] report for the automated tests
- [x] Update **documentation** as appropriate (e.g [README.md](../blob/main/README.md), [Architecture.md](../blob/main/docs/Architecture.md), etc.)
- [x] **Run the demo app** and try the changes
- [x] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^4]

# Reviewer

- [x] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
- [ ] Perform an **ad hoc review**[^5]
- [ ] Review the **automated tests**
- [ ] Review the **manual tests**
- [x] Review the **documentation**, [README.md](../blob/main/README.md), [Architecture.md](/blob/main/docs/Architecture.md), etc. as appropriate
- [ ] **Run the demo app** and try the changes[^6]

[^1]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
[^2]: _While we aim for automated testing of the SDK, some aspects require manual testing. If you had to manually test 
something during development of this pull request, write those steps down._
[^3]: _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
[^4]: _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
[^5]: _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
[^6]: _While the CI server runs the demo app to look for build failures or crashes, humans running the demo app are 
more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

File: sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt [hunk 1]
```diff
@@ -1008,7 +1009,17 @@ class CompactBlockProcessor internal constructor(
...
+
+        /**
+         * This is the same as [SYNC_BATCH_SIZE] but meant to be used in the Zcash sandblasting periods
+         */
+        private const val SYNC_BATCH_SMALL_SIZE = 100
+
+        /**
+         * Known Zcash sandblasting period
+         */
+        private val SANDBLASTING_RANGE = 1_710_000L..2_050_000L
```
Comment (str4d) [inline sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt:1022]: I believe this to be roughly correct. At some point I'll delve back into the chain and figure out more precise boundaries.

**Review by str4d — APPROVED** [review]
utACK 14ac87540ff71148022356bf08fbde7ea94cfbbc

**Comment by str4d** [comment]
> ## Sync speed results

What device were these measured on?

> **New wallet (cca 22K blocks):**
> 
> _Before_:
> 
>     1. 14:15:53.199 - 14:09:02.966 = 6 minutes, 50 seconds, and 233
> 
>     2. 14:24:15.356 - 14:18:32.068 = 5 minutes, 43 seconds, and 288
> 
> 
> _After_:
> 
>     1. 14:01:34.786 - 13:55:54.256 = 5 minutes, 40 seconds, and 530
> 
>     2. 15:18:51.461 - 15:14:26.988 = 4 minut, 24 sekund a 473
> 

1: 340.53 / 410.233 = 83% -> 17% decrease
2: 264.473 / 343.288 = 77% -> 23% decrease

That's pretty significant.

> 
> **Wallet restoration (cca 117K blocks):**
> 
> _Before:_ 16:32:09.854 - 16:02:00.592 = 30 minut, 9 sekund a 262
> 
> _After:_ 15:51:07.254 - 15:22:28.317 = 28 minut, 38 sekund a 937

1718.937 / 1,809.262 = 95% -> 5% decrease

Less significant but still useful.

I'm not entirely surprised that the improvement to sync time is relatively small, as we still have to pay the same block scanning cost.

I also note that both of the above were for post-sandblasting wallets. Once this gets into a Zashi build, I'll test before-vs-after for a wallet with a birthday before sandblasting.

**Review by Milan-Cerovsky — APPROVED** [review]

**Comment by HonzaR** [comment]
> What device were these measured on?

Physical device [Pixel 8](https://www.gsmarena.com/google_pixel_8-12546.php)

> Once this gets into a Zashi build, I'll test before-vs-after for a wallet with a birthday before sandblasting.

I'm going to run such a test. Previously, I only verified that it properly syncs in the sandblasting period with batch size 100, and once outside, it correctly switched to 1000.

**Pull request merged**